kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/utils/FieldNameSanitizer.java

```diff
@@ -21,14 +23,19 @@ public static String sanitizeName(String name) {
   // Note: a.b and a/b will have the same value after sanitization which will cause Duplicate key
   // Exception.
   public static Map<String, Object> replaceInvalidKeys(Map<String, Object> map) {
-    return map.entrySet().stream().collect(Collectors.toMap(
-        (entry) -> sanitizeName(entry.getKey()),
-        (entry) -> {
-          if (entry.getValue() instanceof Map) {
-            return replaceInvalidKeys((Map<String, Object>) entry.getValue());
+    try {
+
+      return map.entrySet().stream().collect(Collectors.toMap(
+          (entry) -> sanitizeName(entry.getKey()),
+          (entry) -> {
+            if (entry.getValue() instanceof Map) {
+              return replaceInvalidKeys((Map<String, Object>) entry.getValue());
+            }
+            return entry.getValue();
           }
-          return entry.getValue();
-        }
-    ));
+      ));
+    } catch (NullPointerException npe) {
```

Comment:
Where exactly is this NPE thrown from? What kind of input data causes this?

Reply:
Here's the stack trace: ```java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:222)
	at java.base/java.util.stream.Collectors.lambda$uniqKeysMapAccumulator$1(Collectors.java:178)
	at java.base/java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
	at java.base/java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1837)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at com.wepay.kafka.connect.bigquery.utils.FieldNameSanitizer.replaceInvalidKeys(FieldNameSanitizer.java:24)
	at com.wepay.kafka.connect.bigquery.BigQuerySinkTask.getRecordRow(BigQuerySinkTask.java:207)
	at com.wepay.kafka.connect.bigquery.BigQuerySinkTask.put(BigQuerySinkTask.java:329)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:583)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:326)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:229)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:198)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:185)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:235)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
```

Reply:
It seems to be happening in `collect` part of `replaceInvalidKeys` so I assumed it's happening after sanitizing the keys. However, it seems to be that you can't have null sanitized name, so right now I am null checking the original key name coming in. I feel like if this was the case the NPE would happen in `sanitizeName` not in `replaceInvalidKeys` but I'm not sure where else the NPE would come from, since I'm pretty sure you can have null values in a map.

Reply:
Can you provide a concrete example of a sink record that would cause this to happen?